### ISSUE TITLE
Allow using pre-existing user and group when installing unprivileged agents

### ISSUE THREAD

**Issue by cmacknz** [elastic/elastic-agent#4585]
Quoting https://github.com/elastic/elastic-agent/issues/3868#issuecomment-2053871719 for a Windows use case for this:

> For example, if I have a service account in active dieectory that works on the system already, can I use that or is elastic creating an account and group locally?
> 
> The use case is integrations that need network access from specific service accounts in active directory to be the service context for filebeat (api, file shares, etc).
> 
> Today system is the only option but this gives me hope for 1 non system which is great, but having custom users would be excellent.
> 
> In the past you could simply install filebeat or winglogbeat with the service account you needed as a service and it would use that user context when it runs.

**Comment by blakerouse** [comment]
This comes back to having an Active Directory based user/group as the service account that Elastic Agent uses to run. 

This would be really nice to have on Windows because it would allow deployed agents to have the correct access and permissions that administrators want all configured in Active Directory. 

Would need to add new installation parameters to allowing passing the group, user and password to configure the service.

**Comment by blakerouse** [comment]
@nimarezainia Sorry I don't understand the ask about tags and pre-defined policy. Can you provide more context?

**Comment by nimarezainia** [comment]
@blakerouse sorry should have been clearer. there has been an ask for a while that we allow the users to tag agents and map those tags to Agent Policies. Some of that is described [here](https://github.com/elastic/security-team/issues/3592). Obviously tagging is a way to group. An extended ask was to allow users in the same Active Directory group to dynamically be placed in a given agent policy. ( like those in Finance dept are always mapped to policy A and Engineering is mapped to policy B dynamically. No matter where they travel etc). Now we have none of these capabilities today. 

Anyway this is way far away from what we are planning so perhaps I am complicating things by asking.

**Comment by pierrehilbert** [comment]
I think we can tag the Agent but to change the applying a policy depending on that tag should probably be done on the Fleet side.

**Comment by jlind23** [comment]
@michalpristas please  sync with Pavel to ensure we're running the right sets of tests for sudo vs non-sudo.
cc @rowlandgeoff @ycombinator

**Comment by ycombinator** [comment]
@nimarezainia @michalpristas has implemented a solution for this issue in https://github.com/elastic/elastic-agent/pull/5988.  However, we still need to write an integration test to exercise the solution.  This, in turn, requires a Windows VM image with Active Directory (there's a PR for this but it's in a private repo).  He's blocked on getting one of those from EngProd at the moment.

**Comment by cmacknz** [comment]
The solution in https://github.com/elastic/elastic-agent/pull/5988 has tests for a custom user and password, we just can't test that it works correctly with Active Directory.

So one option is we merge it and claim the AD support is in technical preview or beta (or just forbid it) and only take it out of technical preview or beta once we have the testing we need in place.

It does look like it is possible to test the AD case manually until we get the ability to automate it, if we merge this before the AD VM image is ready let's make sure QA tests this manually. We should probably do that regardless to see if they can find problems we didn't think of.

I also highlighted on the PR that once we accept existing users as an option to run agent, we will probably be asked for a way to support password rotation and I don't think we have that yet.

**Comment by nimarezainia** [comment]
@kilfoyle I am not sure where we can mention the AD support would be in tech preview. let's discuss.

**Comment by kilfoyle** [comment]
@nimarezainia and @cmacknz  I've opened a docs PR [here](https://github.com/elastic/ingest-docs/pull/1505) with technical preview markers shown in the screencaps. Whatever we need to fix up, just let me know.

**Comment by jlind23** [comment]
@ycombinator according to Craig's comment above this probably should not be considered as blocked anymore right?
Make sure we also involve QA source for manually testing this before we got the automation in place.

**Comment by ycombinator** [comment]
@jlind23 Indeed, it is not blocked anymore and QA has been tagged as well: https://github.com/elastic/elastic-agent/pull/5988#issuecomment-2502464911.
